src/en/kagane/src/eu/kanade/tachiyomi/extension/en/kagane/Kagane.kt

```diff
@@ -0,0 +1,392 @@
+package eu.kanade.tachiyomi.extension.en.kagane
+
+import android.app.Application
+import android.content.SharedPreferences
+import android.os.Handler
+import android.os.Looper
+import android.util.Base64
+import android.view.View
+import android.webkit.CookieManager
+import android.webkit.JavascriptInterface
+import android.webkit.PermissionRequest
+import android.webkit.WebChromeClient
+import android.webkit.WebView
+import androidx.preference.PreferenceScreen
+import androidx.preference.SwitchPreferenceCompat
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import keiyoushi.utils.getPreferencesLazy
+import keiyoushi.utils.parseAs
+import keiyoushi.utils.toJsonString
+import kotlinx.serialization.json.buildJsonObject
+import kotlinx.serialization.json.put
+import okhttp3.Cookie
+import okhttp3.CookieJar
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import rx.Observable
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import kotlin.collections.forEach
+import kotlin.getValue
+import kotlin.text.isNotEmpty
+import kotlin.text.split
+
+class Kagane : HttpSource(), ConfigurableSource {
+
+    override val name = "Kagane"
+
+    private val domain = "kagane.org"
+    private val apiUrl = "https://api.$domain"
+    override val baseUrl = "https://$domain"
+
+    override val lang = "en"
+
+    override val supportsLatest = false
+
+    private val preferences by getPreferencesLazy()
+
+    override val client = network.cloudflareClient.newBuilder()
+        .cookieJar(
```

Comment:
use `CookieInterceptor` lib to set/modify cookies, instead of overriding cookie jar

Reply:
It doesn't do what I need it to.